Getting a crash when running argocd diff

I'm getting a crash when running:
`argocd app diff kibana-backup --local=. --plaintext`

The crash only happens when running `argocd` CLI, the UI does not show any diffs.

Here is the crash:
```                                                   
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xf5802a]

goroutine 1 [running]:
k8s.io/apimachinery/pkg/apis/meta/v1/unstructured.(*Unstructured).GetAPIVersion(0x0, 0x0, 0x0)
        /go/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go:200 +0x3a
k8s.io/apimachinery/pkg/apis/meta/v1/unstructured.(*Unstructured).GroupVersionKind(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go:396 +0x5b
github.com/argoproj/argo-cd/util/kube.GetResourceKey(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/argoproj/argo-cd/util/kube/kube.go:99 +0x66
github.com/argoproj/argo-cd/cmd/argocd/commands.NewApplicationDiffCommand.func1(0xc0005f4b40, 0xc0005993e0, 0x1, 0x3)
        /go/src/github.com/argoproj/argo-cd/cmd/argocd/commands/app.go:690 +0x106b
github.com/spf13/cobra.(*Command).execute(0xc0005f4b40, 0xc000599350, 0x3, 0x3, 0xc0005f4b40, 0xc000599350)
        /go/src/github.com/spf13/cobra/command.go:702 +0x2d3
github.com/spf13/cobra.(*Command).ExecuteC(0xc0005c58c0, 0xc0005c5d40, 0xc0000adb30, 0xc000382ee0)
        /go/src/github.com/spf13/cobra/command.go:783 +0x2dc
github.com/spf13/cobra.(*Command).Execute(0xc0005c58c0, 0xc00020bf88, 0xc0000480b8)
        /go/src/github.com/spf13/cobra/command.go:736 +0x2b
main.main()
        /go/src/github.com/argoproj/argo-cd/cmd/argocd/main.go:14 +0x27
```

I'm using the latest release: 
```
argocd version --plaintext
argocd: v0.12.0-rc6+d5ee07e
  BuildDate: 2019-03-20T22:04:03Z
  GitCommit: d5ee07ef62ce0e79b88b47db0dbd054fe9a6bdf7
  GitTreeState: clean
  GoVersion: go1.11.4
  Compiler: gc
  Platform: linux/amd64
argocd-server: v0.12.0-rc6+d5ee07e
  BuildDate: 2019-03-20T22:05:08Z
  GitCommit: d5ee07ef62ce0e79b88b47db0dbd054fe9a6bdf7
  GitTreeState: clean
  GoVersion: go1.11.4
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: 0.13.1
```

Here are the files I have in my local directory:
`configmap.yaml`, with contents:
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: elk-kibana-backup-config
  namespace: default
  labels:
    role: backup
    app.kubernetes.io/instance: kibana-backup
data:
  boto.cfg: |
    [Credentials]
    gs_service_key_file = /etc/xxx/xxx.json
    default_project_id = xxx-xxx
  backup.sh: |
    export BOTO_PATH=/etc/xxx/boto.cfg
    backup_name="$(date +%Y%m%dT%H%M%S)_saved_objects.json"
    curl -sL 'kibana-svc-int/api/saved_objects?per_page=10000' > "$backup_name"
    gsutil cp -Z "./$backup_name" gs://xxxxxx/xxxxx/
    sleep 5s
    backup_name="$(date +%Y%m%dT%H%M%S)_xxxxx.json"
    elasticdump --input='http://es-ingest:9200/.kibana' --output=./${backup_name}
    gsutil cp -Z "./$backup_name" gs://xxxxxx/xxxxxx/
    sleep 5s
```

and `cronjob.yaml` with contents:
```
# Manual dependencies
# - Create the secret from the svc account
#   - check xxxxx
#   - kubectl xxxx
# - Create the cron job
apiVersion: batch/v1beta1
kind: CronJob
metadata:
  name: elk-kibana-backup
  namespace: default
  labels:
    role: xxxx
    app.kubernetes.io/instance: kibana-backup
spec:
  schedule: "0 0,12 * * *"
  jobTemplate:
    spec:
      template:
        spec:
          containers:
          - name: kibana-backup
            image: 'xxxx/node-elasticdump'
            args:
              - /bin/sh
              - /etc/xxxx/backup.sh
            volumeMounts:
            - name: config
              mountPath: /etc/xxxx
            - name: svckey
              mountPath: /etc/xxxx
              readOnly: true
          volumes:
          - name: config
            configMap:
              name: elk-kibana-backup-config
          - name: svckey
            secret:
              secretName: svckey-xxxx-gs-backups
          restartPolicy: Never
```

Let me know if you need more info!

Edit: seems to have been introduced in `rc5`, `rc4` does not have the crash

--- a/cmd/argocd/commands/app.go
+++ b/cmd/argocd/commands/app.go
@@ -687,9 +687,18 @@ func NewApplicationDiffCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 					err := json.Unmarshal([]byte(res.LiveState), &live)
 					errors.CheckError(err)
 
-					key := kube.GetResourceKey(live)
+					var key kube.ResourceKey
+					if live != nil {
+						key = kube.GetResourceKey(live)
+					} else {
+						var target = &unstructured.Unstructured{}
+						err = json.Unmarshal([]byte(res.TargetState), &target)
+						errors.CheckError(err)
+						key = kube.GetResourceKey(target)
+					}
+
 					if local, ok := localObjs[key]; ok || live != nil {
-						if local != nil && !kube.IsCRD(live) {
+						if local != nil && !kube.IsCRD(local) {
 							err = kube.SetAppInstanceLabel(local, argoSettings.AppLabelKey, appName)
 							errors.CheckError(err)
 						}

